packages/web3-core-types/src/Hex/Hex.js

```diff
@@ -0,0 +1,378 @@
+/*
+    This file is part of web3.js.
+
+    web3.js is free software: you can redistribute it and/or modify
+    it under the terms of the GNU Lesser General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    web3.js is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Lesser General Public License for more details.
+
+    You should have received a copy of the GNU Lesser General Public License
+    along with web3.js.  If not, see <http://www.gnu.org/licenses/>.
+*/
+/**
+ * @file Hex.js
+ * @author Oscar Fonseca <hiro@cehh.io>
+ * @date 2019
+ */
+
+import {isNumber, isString} from 'lodash';
+import utf8 from 'utf8';
+
+export default class Hex {
+    /**
+     * @dev Wrap as object
+     * @param {String} hex
+     *
+     * @constructor
+     */
+    constructor(params) {
+        const requires = ['hex'];
```

Comment:
What exactly is the idea behind this array?

Reply:
For each type, the `requires` array holds the minimum internal properties required to have a valid Type. If by the end of the constructor any of the required properties is still undefined, an error is thrown. It "self tests".